### ISSUE TITLE
Publish in npm repository

### ISSUE THREAD

**Issue by ipritish** [jessety/pm2-installer#14]
This is an excellent solution. Unfortunately our business requirement is such that we can only used published npm packages. Would it be possible to publish it over npm packages?

**Comment by jessety** [comment]
To be clear, this isn't an npm package. It's a collection of scripts, mostly PowerShell, that configure both npm and pm2.

I've considered distributing this on npm in the past, but unfortunately part of the setup process on Windows requires changing the global npm directory and I'm not sure how well that would work for something installed via npm. Additionally, a packaged version wouldn't be able to support the offline installation scripts.

The most realistic path towards at least partial distribution on npm would be to spin off the Windows service component into it's own separate project, while keeping the installer / npm config / offline stuff in this project. I definitely see how the Windows service alone could be helpful to users who already have their own scripts to reconfigure npm and don't need offline support. However, splitting that component out into a new project is a non-trivial amount of work and I can't commit to making that happen in the near future.

I hope this is helpful. I've had to work around some pretty arduous business requirements in the past so I'm definitely sympathetic. I wish I had an easier answer for you!

**Comment by ipritish** [comment]
Makes sense. I ended up copying some of code from the script in my own installation script as a workaround for now. Thanks for replying. I would still like to keep this issue open if you don't mind because if there is any chance of getting it in npm repos it would be very helpful
